pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessagePayloadContextImpl.java

```diff
@@ -68,7 +68,7 @@ public static MessagePayloadContextImpl get(final BrokerEntryMetadata brokerEntr
         final MessagePayloadContextImpl context = RECYCLER.get();
         context.consumerEpoch = consumerEpoch;
         context.brokerEntryMetadata = brokerEntryMetadata;
-        context.messageMetadata = messageMetadata;
+        context.messageMetadata.copyFrom(messageMetadata);
```

Comment:
This may be problematic because the filter may want to hold a reference to this object (I don't know why but you know users sometimes do silly things)